content/app/ui/popups/goto-file-popup.js

```diff
@@ -93,10 +117,44 @@ define("content/app/ui/popups/goto-file-popup", function(require, exports, modul
 		 * Trigger submit event.
 		 */
 		_triggerSubmit : function () {
-			var event = new Event(Event.SUBMIT, this._input.value);
+
+			var inputValue = this._input.value;
+			var choosenEntry = null;
+			var entries = this._entries;
+
+			for (var i = 0; i < entries.length; i++) {
+				if (inputValue === entries[i].toString()) {
+					choosenEntry = entries[i];
+					break;
+				}
+			}
+
+			var fileWrapper = choosenEntry ? choosenEntry.fileElementWrapper : null;
```

Comment:
This component should not know anything about entry property instead that it have toString() method.
So your event should keep choosenEntry instead fileWrapper. And component that created this popup will know what is type of data in event.
Take a look at goto-file-command.js line 40, entries are created out of popup.